app/actions/local/channel.ts

```diff
@@ -486,3 +491,35 @@ export async function showUnreadChannelsOnly(serverUrl: string, onlyUnreads: boo
         prepareRecordsOnly: false,
     });
 }
+
+export const updateDmGmDisplayName = async (serverUrl: string) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    try {
+        const currentUserId = await getCurrentUserId(database);
+        if (!currentUserId) {
+            return {error: 'The current user id could not be retrieved from the database'};
+        }
+
+        const channels = await queryUserChannelsByTypes(database, currentUserId, ['G', 'D']).fetch();
+        const userIds = channels.reduce((acc: string[], ch) => {
+            const uid = getUserIdFromChannelName(currentUserId, ch.name);
```

Comment:
maybe before calling `getUserIdFromChannelName` you can check that the channel is a DM